From 28f23b62d72091e5b8e2ee8d81b6753ac0409380 Mon Sep 17 00:00:00 2001
From: Jon Turney <jon.turney@dronecode.org.uk>
Date: Sat, 5 Mar 2016 12:56:22 +0000
Subject: [PATCH 04/23] Build minidump_upload, sym_upload for all targets

Rather than duplicating these tools for any target which has libcurl

- Introduce src/tools/common/ and move minidump_upload, sym_upload there
from src/tools/linux/
- Move linux/http_upload.cc,.h up to src/common/ from src/common/linux/

Signed-off-by: Jon Turney <jon.turney@dronecode.org.uk>
---
 Makefile.am                                        | 29 +++++++++++++---------
 src/common/{linux => }/http_upload.cc              |  2 +-
 src/common/{linux => }/http_upload.h               |  0
 .../{linux => common}/symupload/minidump_upload.cc |  2 +-
 .../{linux => common}/symupload/sym_upload.cc      |  2 +-
 5 files changed, 20 insertions(+), 15 deletions(-)
 rename src/common/{linux => }/http_upload.cc (99%)
 rename src/common/{linux => }/http_upload.h (100%)
 rename src/tools/{linux => common}/symupload/minidump_upload.cc (99%)
 rename src/tools/{linux => common}/symupload/sym_upload.cc (99%)

diff --git a/Makefile.am b/Makefile.am
index 0ee1249..def5be7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -337,6 +337,12 @@ bin_PROGRAMS += \
 	src/processor/minidump_stackwalk
 endif !DISABLE_PROCESSOR
 
+if !DISABLE_TOOLS
+bin_PROGRAMS += \
+	src/tools/common/symupload/minidump_upload \
+	src/tools/common/symupload/sym_upload
+endif
+
 if LINUX_HOST
 EXTRA_PROGRAMS += \
 	src/client/linux/linux_dumper_unittest_helper
@@ -347,9 +353,7 @@ if !DISABLE_TOOLS
 bin_PROGRAMS += \
 	src/tools/linux/core2md/core2md \
 	src/tools/linux/dump_syms/dump_syms \
-	src/tools/linux/md2core/minidump-2-core \
-	src/tools/linux/symupload/minidump_upload \
-	src/tools/linux/symupload/sym_upload
+	src/tools/linux/md2core/minidump-2-core
 if X86_HOST
 bin_PROGRAMS += \
 	src/tools/mac/dump_syms/dump_syms_mac
@@ -440,6 +444,16 @@ else
 LOG_DRIVER = $(top_srcdir)/autotools/test-driver
 endif
 
+src_tools_common_symupload_minidump_upload_SOURCES = \
+	src/common/http_upload.cc \
+	src/tools/common/symupload/minidump_upload.cc
+src_tools_common_symupload_minidump_upload_LDADD = -lcurl
+
+src_tools_common_symupload_sym_upload_SOURCES = \
+	src/common/http_upload.cc \
+	src/tools/common/symupload/sym_upload.cc
+src_tools_common_symupload_sym_upload_LDADD = -lcurl
+
 if LINUX_HOST
 src_client_linux_linux_dumper_unittest_helper_SOURCES = \
 	src/client/linux/minidump_writer/linux_dumper_unittest_helper.cc
@@ -573,15 +587,6 @@ src_tools_linux_md2core_minidump_2_core_SOURCES = \
 	src/common/linux/memory_mapped_file.cc \
 	src/tools/linux/md2core/minidump-2-core.cc
 
-src_tools_linux_symupload_minidump_upload_SOURCES = \
-	src/common/linux/http_upload.cc \
-	src/tools/linux/symupload/minidump_upload.cc
-src_tools_linux_symupload_minidump_upload_LDADD = -lcurl
-
-src_tools_linux_symupload_sym_upload_SOURCES = \
-	src/common/linux/http_upload.cc \
-	src/tools/linux/symupload/sym_upload.cc
-src_tools_linux_symupload_sym_upload_LDADD = -lcurl
 
 src_tools_mac_dump_syms_dump_syms_mac_SOURCES = \
 	src/common/dwarf_cfi_to_module.cc \
diff --git a/src/common/linux/http_upload.cc b/src/common/http_upload.cc
similarity index 99%
rename from src/common/linux/http_upload.cc
rename to src/common/http_upload.cc
index 6670000..1d57cf3 100644
--- a/src/common/linux/http_upload.cc
+++ b/src/common/http_upload.cc
@@ -27,7 +27,7 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#include "common/linux/http_upload.h"
+#include "common/http_upload.h"
 
 #include <assert.h>
 #include <curl/curl.h>
diff --git a/src/common/linux/http_upload.h b/src/common/http_upload.h
similarity index 100%
rename from src/common/linux/http_upload.h
rename to src/common/http_upload.h
diff --git a/src/tools/linux/symupload/minidump_upload.cc b/src/tools/common/symupload/minidump_upload.cc
similarity index 99%
rename from src/tools/linux/symupload/minidump_upload.cc
rename to src/tools/common/symupload/minidump_upload.cc
index 19f1745..ed0dd5f 100644
--- a/src/tools/linux/symupload/minidump_upload.cc
+++ b/src/tools/common/symupload/minidump_upload.cc
@@ -40,7 +40,7 @@
 
 #include <string>
 
-#include "common/linux/http_upload.h"
+#include "common/http_upload.h"
 #include "common/using_std_string.h"
 
 using google_breakpad::HTTPUpload;
diff --git a/src/tools/linux/symupload/sym_upload.cc b/src/tools/common/symupload/sym_upload.cc
similarity index 99%
rename from src/tools/linux/symupload/sym_upload.cc
rename to src/tools/common/symupload/sym_upload.cc
index 2f9a73c..4f50e47 100644
--- a/src/tools/linux/symupload/sym_upload.cc
+++ b/src/tools/common/symupload/sym_upload.cc
@@ -49,7 +49,7 @@
 #include <string>
 #include <vector>
 
-#include "common/linux/http_upload.h"
+#include "common/http_upload.h"
 #include "common/using_std_string.h"
 
 using google_breakpad::HTTPUpload;
-- 
2.7.0

